addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java

```diff
@@ -0,0 +1,1150 @@
+            method.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
+                    new DefaultHttpMethodRetryHandler(3, false));
+            // Set the right header
+            method.setRequestHeader("Accept", "application/json");
+            method.addRequestHeader("User-Agent", "TeleHeater/2.2.3");
+            method.setRequestEntity(new ByteArrayRequestEntity(data));
+
+            try {
+                rCode = client.executeMethod(method);
+
+            } catch (Exception e) {
+                logger.error("Failed to send data {}", e);
+
+            } finally {
+                // Release the connection.
+                method.releaseConnection();
+            }
+            return rCode;
+        }
+    }
+
+    /**
+     * This function does the decoding for a new message from the device
+     *
+     */
+    public String decodeMessage(byte[] encoded) {
+        String retString = null;
+        byte[] decodedB64 = null;
+
+        try {
+            decodedB64 = Base64.decodeBase64(encoded);
+        } catch (Exception e) {
+            logger.error("Message is not in valid Base64 scheme: {}", e.getMessage());
+            return null;
+        }
+        try {
+            /* Check whether the length of the decryptData is NOT multiplies of 16 */
+            if ((decodedB64.length & 0xF) != 0) {
+                /* Return the data */
+                retString = new String(decodedB64, device.getCharSet());
+                return retString;
+            }
+            // --- create cipher
+            final Cipher cipher = Cipher.getInstance("AES/ECB/NoPadding");
+            cipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(device.getCryptKeyPriv(), "AES"));
+            final byte[] decryptedData = cipher.doFinal(decodedB64);
+            byte[] decryptedDataWOZP = removeZeroPadding(decryptedData);
+            retString = new String(decryptedDataWOZP, device.getCharSet());
+            return retString;
+        } catch (UnsupportedEncodingException | GeneralSecurityException e) {
+            // failure to authenticate
+            logger.error("Exception on encoding: {}", e);
+            return null;
+        }
+    }
+
+    /**
+     * This function does the encoding for a new message to the device
+     *
+     */
+    public byte[] encodeMessage(String data) {
+        byte[] encryptedDataB64 = null;
+
+        try {
+            // --- create cipher
+            byte[] bdata = data.getBytes(device.getCharSet());
+            final Cipher cipher = Cipher.getInstance("AES/ECB/NoPadding");
+            cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(device.getCryptKeyPriv(), "AES"));
+            int bsize = cipher.getBlockSize();
+            logger.debug("Add Padding, encrypt AES and B64..");
+            final byte[] encryptedData = cipher.doFinal(addZeroPadding(bdata, bsize, device.getCharSet()));
+            try {
+                encryptedDataB64 = Base64.encodeBase64(encryptedData);
```

Comment:
you may return here directly and `return null` at the end of the method in case something goes wrong.